xapian-letor/m4-macros/xapianletor.m4

```diff
@@ -0,0 +1,172 @@
+# Get XAPIAN_CXXFLAGS, XAPIAN_LIBS, and XAPIAN_VERSION from xapianletor-config and
+# AC_SUBST() them.
+
+# serial 17
+
+# AC_PROVIDE_IFELSE(MACRO-NAME, IF-PROVIDED, IF-NOT-PROVIDED)
+# -----------------------------------------------------------
+# If this macro is not defined by Autoconf, define it here.
+m4_ifdef([AC_PROVIDE_IFELSE],
+	 [],
+	 [m4_define([AC_PROVIDE_IFELSE],
+		 [m4_ifdef([AC_PROVIDE_$1],
+			   [$2], [$3])])])
+
+# XO_LIB_XAPIANLETOR([ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND[ ,XAPIANLETOR-CONFIG]]])
+# --------------------------------------------------------
+# AC_SUBST-s XAPIAN_CXXFLAGS, XAPIAN_LIBS, and XAPIAN_VERSION for use in
+# Makefile.am
+#
+# If ACTION-IF-FOUND and ACTION-IF-NOT-FOUND are both unset, then an
+# appropriate AC_MSG_ERROR is used as a default ACTION-IF-NOT-FOUND.
+# This allows XO_LIB_XAPIANLETOR to be used without any arguments in the
+# common case where Xapian is a requirement (rather than optional).
+#
+# XAPIANLETOR-CONFIG provides the default name for the xapianletor-config script
+# (which the user can override with "./configure XAPIANLETOR_CONFIG=/path/to/it").
+# If unset, the default is xapianletor-config.  Support for this third parameter
+# was added in Xapian 1.1.3.
+AC_DEFUN([XO_LIB_XAPIANLETOR],
+[
+  AC_ARG_VAR(XAPIANLETOR_CONFIG, [Location of xapianletor-config (default:] ifelse([$3], [], xapianletor-config, [$3]) [on PATH)])
+  dnl AC_PATH_PROG ignores an existing user setting of XAPIANLETOR_CONFIG unless it
+  dnl has a full path, so add special handling for such cases.
+  xapian_config_to_check_for="ifelse([$3], [], xapianletor-config, [$3])"
```

Comment:
I think this is only a local variable so we'll get away with this having the same name as in `xapian.m4`, but it would be less confusing to call this `xapianletor_config_to_check_for` or similar.  Or rename both to something generic like `config_script_to_check_for` and keep the two sources more aligned.

Reply:
fixed in a fixup commit.